Code examples of filters' usage, issue #289

alexkravin opened the pull request:
According to #289

Added examples of suppressed java source code due to specified filter options in config.xml

Examples were added only to filters which work with Java code. (because there're also XML filter, or suppressions loader, which are already documented and don't need any code illustrations.)

romani commented:
Travis found a problem, please resolve.

alexkravin commented:
Now ok, it was very strange violation.. 'Tag <br> isn't closed with </br>'

romani commented:
problem again, please run "mvn clean verify" on your side.

romani commented:
make sure that you verify how site is generated my "mvn clean package -Pdistro -Dlinkcheck.skip=true".

Blind changes to files are not enough!

romani commented:
please generate web site and review each changes and make them readable and pretty look.
current formatting is not acceptable.

alexkravin commented:
here's current config.html:
http://alexkravin.github.io/checkstyle/config.html

romani commented:
good :) , thanks for pushing results to hosting :).

but:

1) please add example for "<property name="offCommentFormat" value="BEGIN GENERATED CODE"/>"

2) that line is too long and wrapped on second row "//will warn here as constant's name doesn't match
the pattern "^[A-Z][A-Z0-9]*$""
, but that code chunk have to be compilable

3) please make code example for "<property name="offCommentFormat" value="UNUSED OFF\: (\w+)"/>"
and all other xml configurations !!!!!!
That page need to be clear for new Checkstyle user ("for dummies"). 

4) I do not understand mix of code and xml config exmaples for "SuppressWarningsFilter", please review.

alexkravin commented:
```
1) please add example for ""
```

can you specify these parts?

romani commented:
Message was updated

alexkravin commented:
Added examples to each case of xml declaration of filter:
http://alexkravin.github.io/checkstyle/config.html

romani commented:
1) 
> As another example, imagine that a configuration contains the following: 

as another example to suppress Check by module id:

2)
please add following examples : 

```
    <!-- Suppress checks for hidden files and folders: -->
    <suppress files="[/\\]\..+" checks=".*"/>

    <!-- Suppress checks for Maven-generated code: -->
    <suppress files="[/\\]target[/\\]" checks=".*"/>

    <!-- Suppress checks for archives, classes and other binary files: -->
    <suppress files=".+\.(?:jar|zip|war|class|tar|bin)$" checks=".*"/>

    <!-- Suppress checks for image files: -->
    <suppress files=".+\.(?:png|gif|jpg|jpeg)$" checks=".*"/>

    <!-- Suppress checks for non-java text files. -->
    <suppress files=".+\.(?:txt|xml|csv|sh|thrift|html|sql|eot|ttf|woff|css|png)$" checks=".*"/>

    <!-- Suppress all checks generated sources. -->
    <suppress checks=".*" files="com[\\/]mycompany[\\/]app[\\/]gen[\\/]" />

    <!-- Suppress FileLength on International tests in certain folder. -->
    <suppress checks="FileLength" files="com[\\/]mycompany[\\/]app[\\/].*IT.java" />
```

3) 
> To configure a filter ....

By that sentence all examples are started, but example of code is following....
please swap example of code with example of config.

4) 
> private static void foo(int a, int b) // UNUSED OFF: b

no pair comments

5) 
> To configure a filter so that CSOFF: regexp and CSN: regexp mark a matching check: 

Phrases like that should be substituded with human friendly description - "To configure a filter so that name of suppressed check mentioned in comment"

6) 
> 
```   <property name="offCommentFormat" value="CHECKSTYLE_OFF: ALL"/> ```

please make it as ALMOST_ALL .

7)
> //UNUSED OFF: someInt
public static void doit1(int someInt) //will warn here

what is difference with example from above about "UNUSED OFF: b"

8)
> SuppressWarningsFilter

please fix indentation in examples.
Please describe how name of check should be written in annotation (lowercase only or .... ??)
please give one more example, with two check suppression on the same line.

9)
>
```
. . .
catch (RuntimeException re) { ... }
catch (Throwable th) { ... } // OK to catch RuntimeException here
. . .
```
 please make example is more realistic like :  
```
catch (RuntimeException re) { 
       // OK to catch RuntimeException here
}
catch (Throwable th) { ... } 

```

10) 
please add example 
```
 <!-- Allow suppress one or more Checks(separated by "|") and but demand comment no less than 14 symbols--> 
  <module name="SuppressWithNearbyCommentFilter">
    <property name="commentFormat" value="@cs.suppress \[(\w(\|\w)*)\] \w[-\.'`,:;\w ]{14,}"/>
    <property name="checkFormat" value="$1"/>
    <property name="influenceFormat" value="1"/>
  </module>

```

11)
> //Will be skipped in audit

// No violation events will be reported

alexkravin commented:
1) Changed

2) Added example for each case

3) In each such case swapped: config goes first, code example - second

4) Added pair comment

5) Changed

6) Done

7) No difference, I don't know why somebody added same example twice, removed

8) Updated for each point

9) Refactored

10) Added config example and corresponding code example

11) Changed

http://alexkravin.github.io/checkstyle/config.html

romani commented:
merged as FF with additional changes from myself (to speedup apply for this PR).

romani commented:
thanks for generated web site hosting on your github.io it helps a lot !
